### PR TITLE
Add payment gateway suggestion unit tests

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes (part of) #7122 

* Adds tests around the base init class.
* Fixes up some remaining semantic issues.

### Detailed test instructions:

1. Make sure tests pass and are comprehensive.

### CODE REVIEW

**Comment by samueljseay** [comment]
Just noting, looks like there's a bug in the unit tests.

**Comment by joshuatf** [comment]
Thanks, @samueljseay.  I suspect there may be a bug in a rule evaluator based on the previous error code, but I've updated the default gateways in these tests which has removed the bug for the time being.

File: tests/features/payment-gateway-suggestions/payment-gateway-suggestions.php
Comment (louwie17) [inline tests/features/payment-gateway-suggestions/payment-gateway-suggestions.php:177]: Maybe also check that the `$localized_specs` has only one item, as this test would still pass if both of the locales were returned above.

File: tests/features/payment-gateway-suggestions/payment-gateway-suggestions.php
Comment (louwie17) [inline tests/features/payment-gateway-suggestions/payment-gateway-suggestions.php:74]: Could we make this one a little more specific then just not empty, maybe comparing directly with `DefaultPaymentGateways::get_all()`?

File: src/Features/PaymentGatewaySuggestions/Init.php
Comment (louwie17) [inline src/Features/PaymentGatewaySuggestions/Init.php:127]: I assume this isn't relevant anymore?

**Review by louwie17 — COMMENTED** [review]
Thanks for adding tests @joshuatf, I left a couple small suggestions and one question.

File: tests/features/payment-gateway-suggestions/payment-gateway-suggestions.php
Comment (joshuatf) [reply tests/features/payment-gateway-suggestions/payment-gateway-suggestions.php]: That's a much better assertion- thanks!

File: tests/features/payment-gateway-suggestions/payment-gateway-suggestions.php
Comment (joshuatf) [reply tests/features/payment-gateway-suggestions/payment-gateway-suggestions.php:180]: The array indexes here are referring to the first spec, not the locale, so this would not pass if both were returned, but I'm not opposed to adding a count check here.

File: src/Features/PaymentGatewaySuggestions/Init.php
Comment (joshuatf) [reply src/Features/PaymentGatewaySuggestions/Init.php:127]: Correct, fields are now handled directly from the payment gateway and passed via the REST API.

**Comment by joshuatf** [comment]
Thanks for the feedback @louwie17!  Handled in https://github.com/woocommerce/woocommerce-admin/pull/7142/commits/bb28d7fd2e816825623e404dac3d8c4395832693

**Review by louwie17 — APPROVED** [review]
> Thanks for the feedback @louwie17! Handled in [bb28d7f](https://github.com/woocommerce/woocommerce-admin/commit/bb28d7fd2e816825623e404dac3d8c4395832693)

Thanks for updating it, I ran the tests locally and every thing passes :) Good to merge once the merge conflicts are fixed 🚀

**Pull request merged**
